### PR TITLE
Issue #572 S2326 shouldn't reports on interface implementations

### LINKED ISSUE #572: S2326 shouldn't reports on interface implementations

### PR DESCRIPTION
**Pull request by tvsonar**
Fixes #572

### CODE REVIEW

**Comment by dbolkensteyn** [comment]
There's a merge conflict now on this one

**Comment by tvsonar** [comment]
@dbolkensteyn I did the rebase it should be okay now.

File: src/SonarLint.CSharp/Rules/GenericTypeParameterUnused.cs [hunk 1]
```diff
@@ -122,6 +106,48 @@ public override void Initialize(AnalysisContext context)
             });
         }
 
+        private static TypeParameterHelper GetTypeParameterHelper(MethodDeclarationSyntax methodDeclaration, ClassDeclarationSyntax classDeclaration)
+        {
+            return classDeclaration == null
```
Comment (dbolkensteyn) [inline src/SonarLint.CSharp/Rules/GenericTypeParameterUnused.cs]: @tvsonar this looks unusual due to the indentation, perhaps it'd be better to use an if (the previous code was more clear IMO)

File: src/SonarLint.CSharp/Rules/MethodParameterUnused.cs [hunk 2]
```diff
@@ -65,7 +65,8 @@ public override void Initialize(AnalysisContext context)
                     var methodSymbol = cbc.OwningSymbol as IMethodSymbol;
                     if (methodDeclaration == null ||
                         methodSymbol == null ||
-                        !IsMethodCandidate(methodSymbol, cbc.SemanticModel.Compilation))
+                        !IsMethodChangeable(methodSymbol, cbc.SemanticModel.Compilation) ||
```
Comment (dbolkensteyn) [inline src/SonarLint.CSharp/Rules/MethodParameterUnused.cs]: @tvsonar as on the other similar PR, at some point we'll probably want to move this to a Helpers folder to make such helpers way easier to find

**Comment by dbolkensteyn** [comment]
LGTM

**Pull request merged**